app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -9,12 +9,15 @@ class TwoFactorAuthenticationSetupController < ApplicationController
 
     def index
       two_factor_options_form
+      @sign_up_mfa_selection_order_bucket = AbTests::SIGN_UP_MFA_SELECTION.bucket(request.ip)
```

Comment:
Would it be preferable to use a value that is less likely to change like the user or session ID?

Reply:
yea i just used this for testing but session id is a better fit.

Reply:
Yeah, I'd personally opted for IP address in #8112, but only because we don't have anything else specific to the user to work with on the Sign In page.

Reply:
I think IP address is bad because somebody on a mobile phone could be moving and switch cell towers, which I think would affect their IP

Reply:
yup im using session.id since im logged in

Reply:
Yeah in my local branch for #8112 I'd also switched to `session.id` since it exists immediately without a sign-in, and probably more stable, since we want to be logging the bucket details in events which might be far detached from the initial assignment (e.g. account creation completed).

Though, now that I think of it, the account creation completed could happen in a separate session, since that'd be after the user clicked the link in their email, which they might do on a different device altogether...  🤔